src/abstractdataframe/sort.jl

```diff
@@ -682,16 +701,91 @@ function Base.sort!(df::AbstractDataFrame, cols=All();
                     lt::Union{Function, AbstractVector{<:Function}}=isless,
                     by::Union{Function, AbstractVector{<:Function}}=identity,
                     rev::Union{Bool, AbstractVector{Bool}}=false,
-                    order::Union{Ordering, AbstractVector{<:Ordering}}=Forward)
+                    order::Union{Ordering, AbstractVector{<:Ordering}}=Forward,
+                    checkunique::Bool=false)
 
     # exclude AbstractVector as in that case cols can contain order(...) clauses
     if cols isa MultiColumnIndex && !(cols isa AbstractVector)
         cols = index(df)[cols]
     end
     ord = ordering(df, cols, lt, by, rev, order)
     _alg = Sort.defalg(df, ord; alg=alg, cols=cols)
+    checkunique && _perform_uniqueness_checks(df, cols, lt, by, order)
     return sort!(df, _alg, ord)
 end
 
-Base.sort!(df::AbstractDataFrame, a::Base.Sort.Algorithm, o::Base.Sort.Ordering) =
+function Base.sort!(df::AbstractDataFrame, a::Base.Sort.Algorithm,
```

Comment:
we should have `checkunique` kwarg also for this method.

Reply:
I did consider it. I settled on not adding it since,
1. It would go unused. At that moment in the code, the uniqueness checks have already been made at `sort(df, cols, alg,...)`
2. At that moment we have no explicit handle on `by`, `lt` and so on. We have access to an `Ordering`, which we could check. But once again, those checks have already happened in the caller
3. The function's only call is by `sort(df, cols, alg,...)`

Reply:
> The function's only call is by `sort(df, cols, alg,...)`

This is the point. The issue is that `sort!` is public function. So it could be called by user. If you prefer we can throw an error in this function if `checkunique=true` is passed informing that it is not supported for this method.

Reply:
Ohh, I see the Issue. I can implement the checks for that function as well, no worries

Reply:
Sure. Thank you!

DataFrames.jl is a widely used package so we need to make sure that we make it maximally robust and future proof.
The unfortunate consequence is that even seemingly conceptually simple PRs are not that simple.